lib/Backends/CPU/FunctionSpecializer.cpp

```diff
@@ -343,7 +339,34 @@ class FunctionSpecializer {
 } // namespace
 
 void LLVMIRGen::performSpecialization() {
-  FunctionSpecializer FuncSpecializer(llmodule_->getFunction("main"));
+  // Specialize all invocations of main functions with constant parameters to
+  // produce a better optimized code.
+  auto *func = getModule().getFunction("jitmain");
+  while (func) {
+    // Call of a "main" function is always the last instruction before return.
+    auto instrBeforeTerminator =
+        std::prev(func->back().getTerminator()->getIterator());
+    auto *call = dyn_cast<llvm::CallInst>(instrBeforeTerminator);
+    if (!call)
+      break;
+    // Specialize only calls of all "main" functions.
+    func = call->getCalledFunction();
+    if (!func || !func->getName().startswith("main"))
```

Comment:
What do you mean starts with "main"? Why do we have multiple main functions?

Reply:
We have one single "main" function invoked from "jitmain" or from the AOT bundle entry point. But this "main" function sometimes becomes too big and so we split/cut it into multiple functions in this PR, so that each one is not too big to swallow for the LLVM optimizer. Each part is called "main.0", "main.1", etc (but we can change this).